pt/lessons/basics/comprehensions.md

```diff
@@ -0,0 +1,114 @@
+---
+layout: page
+title: Comprehensions
+category: basics
+order: 13
+lang: pt
+---
+
+Comprehensions de listas são um 'syntactic sugar' (uma forma mais simples de escrever) para realizar loops em Enumerables em Elixir. Nessa lição veremos como podemos fazer iterações e gerar os resultados utilizando comprehensions.
+
+{% include toc.html %}
+
+## Básico
+
+Em alguns casos comprehensions podem ser usadas para produzir código mais conciso para fazer iterações com `Enum` e `Stream`. Vamos começar olhando para uma comprehension simples e então observar suas várias partes:
+
+```elixir
+iex> list = [1, 2, 3, 4, 5]
+iex> for x <- list, do: x*x
+[1, 4, 9, 16, 25]
+```
+
+A primeira coisa que observamos é o uso de `for` e um generator (gerador). O que é um generator? Generators são as expressões `x <- [1, 2, 3, 4]` encontradas em comprehensions de listas. Eles são responsáveis por gerar o próximo valor.
```

Comment:
Would be good to change here too, or remove `de listas` or add `listas de comprehensions`